readline/src/main/java/org/aesh/readline/Buffer.java

```diff
@@ -54,10 +57,12 @@
     private boolean isPromptDisplayed = false;
     private boolean deletingBackward = true;
 
+    private final CursorLocator locator;
```

Comment:
Do we need to add CursorLocator as a field? Would it work to just create a new instance of a CursorLocator in the getCursorLocator() method?